metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DepublishRecordIdDao.java

```diff
@@ -157,9 +157,8 @@ public Integer deletePendingRecordIds(String datasetId, Set<String> recordIds)
     query.field(DepublishRecordId.DEPUBLICATION_STATUS_FIELD)
         .equal(DepublicationStatus.PENDING_DEPUBLICATION);
 
-    return ExternalRequestUtil.retryableExternalRequestConnectionReset(() -> {
-      return morphiaDatastoreProvider.getDatastore().delete(query).getN();
-    });
+    return ExternalRequestUtil.retryableExternalRequestConnectionReset(
+        morphiaDatastoreProvider.getDatastore().delete(query)::getN);
```

Comment:
Ah, I don't think this is right! This evaluates the delete query immediately, upon creation of the lambda! Not when it is called. I think you should use the ()-> notation here.

Reply:
Ooh yes! Good catch!